.github/workflows/apidoc.yml

```
@@ -0,0 +1,30 @@
+on:
+  push:
+    branches:
+      - develop
+    paths-ignore:
+      - 'docs/**'
+      - 'config/**'
+      - '**.md'
+      - '.dockerignore'
+      - 'docker/**'
+      - '.gitignore'
+
+name: Build API docs
+
+env:
+  RUST_TOOLCHAIN: nightly-2023-02-26
+
+jobs:
+  apidoc:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - uses: dtolnay/rust-toolchain@master
+      with:
+        toolchain: ${{ env.RUST_TOOLCHAIN }}
+    - run: cargo doc
```

```suggestion
    - run: cargo doc --workspace --no-deps --document-private-items
```

These flags: ignore deps, and only generate documents for all items in our workspace.

---

hmm.. I was adding deps doc intentionally so we can find doc for all APIs available.

---

exclude deps to speedup

---

Docs of our dependency are available in `docs.rs`, we may not need to replicate one.